File: Foundation/PatternLecture/l002Pattern.cpp
void printtriangle(int row)
{
    int nst=1;   //no of stars
    for(int r=1;r<=row;r++)
      {
          for(int cst=1;cst<=nst;cst++)
          {
              cout<<"*\t";
          }
          cout<<endl;
          nst++;
      }
}
void printReversetriangle(int row)
{
    int nst=row;   //no of stars
    for(int r=1;r<=row;r++)
      {
          for(int cst=nst;cst>=1;cst--)
          {
              cout<<"*\t";
          }
          cout<<endl;
          nst--;
      }
}
void printMirrortriangle2(int row)
{
    int nst=row;   //no of stars
    int nsp=0;
    for(int r=1;r<=row;r++)
      {
          for(int csp=1;csp<=nsp;csp++)
          {
              cout<<"\t";
          }
          for(int cst=nst;cst>=1;cst--)
          {
              cout<<"*\t";
          }
          cout<<endl;
          nst--;
          nsp++;
      }
}
void Pattern7(int row)
{
    int nsp=1;
    for(int r=1;r<=row;r++)
    {
        for(int csp=1;csp<nsp;csp++)
        {
            cout<<"\t";
        }
        cout<<"*\t"<<endl;
        nsp++;
        
    }

}

void Pattern10(int row)
{
    int nsp1=row/2;
    int nsp2=-1;
    for(int r=1;r<=row;r++){
        for(int csp=1;csp<=nsp1;csp++)
        {
            cout<<"\t";
        }
        cout<<"*\t";
        for(int csp=1;csp<=nsp2;csp++)
        {
            cout<<"\t";
        }
        if(r==1 || r==row)
        {
        cout<<endl;
        }else{
              cout<<"*\t"<<endl;
        }
        if(r<=row/2)
        {
            nsp2+=2;
            nsp1--;
        }
        else{
            nsp2-=2;
            nsp1++;
        }
    }
}
void pattern12(int row)
{
    int nst=1;
    int a=0,b=1;
     int count=0;
    for(int r=1;r<=row;r++)
    {
        for(int cst=1;cst<=nst;cst++)
        {
            cout<<a<<"\t";
            count=a+b;
            a=b;
            b=count;
        }
        cout<<endl;
        nst++;
    }
}

void pattern16(int row)
{    
   
    int nst=1;
    int nsp=2*row-3;
    for(int r=1;r<=row;r++)
    {
         int count=1;
        for(int cst=1;cst<=nst;cst++)
        {
            cout<<count<<"\t";
            count++;
        }
        for(int csp=1;csp<=nsp;csp++)
        {
            cout<<"\t";
        }
        count--;
        int cst=1;
        if(r==row)
        {
        count--;
        cst=2;
        }
        while(cst<=nst)
        {
          cout<<count<<"\t";
          cst++;
          count--;
        }
        nsp-=2;
        nst++;
        cout<<endl;
    }
}

void pattern14(int x)
{
    for(int i=1;i<=10;i++)
    {
        cout<<x<<" * "<<i<<" = "<< x*i<<endl;
    }
}

void pattern17(int row)
    {
        int nsp=row/2,nst=1,mnst=row;
        for(int r=1;r<=row;r++)
        {   
            if(r!=row/2+1)
            {
            for(int csp=0;csp<nsp;csp++)
            {
                cout<<"\t";
            }
            
            for(int cst=1;cst<=nst;cst++)
            {
                 cout<<"*\t";
            }
            }
            else{
                for(int cst=1;cst<=mnst;cst++)
                {
                    cout<<"*\t";
                }
            }
            if(r<=row/2)
            {
                nst++;
            }
            else{
                nst--;
            }
            cout<<endl;
        }
    }

File: Foundation/PatternLecture/l002Pattern.java
 public static void main(String[] args) {
        Scanner scn = new Scanner(System.in);

        // write ur code here
        int n=scn.nextInt();
        printtriangle(n);

    }
   public static void printtriangle(int row)
{
    int nst=1;   //no of stars
    for(int r=1;r<=row;r++)
      {
          for(int cst=1;cst<=nst;cst++)
          {
              System.out.print("*\t");
          }
           System.out.println();
          nst++;
      }
}
  public static void printReversetriangle(int row)
{
    int nst=row;   //no of stars
    for(int r=1;r<=row;r++)
      {
          for(int cst=nst;cst>=1;cst--)
          {
              System.out.print("*\t");
          }
           System.out.println();
          nst--;
      }
}
 public static void printMirrortriangle(int row)
{
    int nst=1;   //no of stars
    int nsp=row-1;
    for(int r=1;r<=row;r++)
      {
          for(int csp=1;csp<=nsp;csp++)
          {
              System.out.print("\t");
          }
          for(int cst=1;cst<=nst;cst++)
          {
              System.out.print("*\t");
          }
           System.out.println();
          nst++;
          nsp--;
      }
}

public static void Pattern5(int row)
{
    int nst=1;
    int nsp=row/2;
    for(int r=1;r<=row;r++)
    {
        for(int csp=1;csp<=nsp;csp++)
        {
            System.out.print("\t");
        }
        for(int cst=1;cst<=nst;cst++)
        {
            System.out.print("*\t");
        }
        if(r<=row/2)
        {
            nst+=2;
            nsp--;
        }
        else{
            nst-=2;
            nsp++;
        }
        System.out.println();
    }
}
public static void Pattern6(int row)
{
    int nst=row/2+1;
    int nsp=1;
    for(int r=1;r<=row;r++)
    {
        for(int cst=1;cst<=nst;cst++)
        {
            System.out.print("*\t");
        }
        for(int csp=1;csp<=nsp;csp++)
        {
            System.out.print("\t");
        }
        for(int cst=1;cst<=nst;cst++)
        {
            System.out.print("*\t");
        }
        if(r<=row/2)
        {
            nst--;
            nsp+=2;
        }
        else{
            nst++;
            nsp-=2;;
        }
        System.out.println();
    }
}

   public static void Pattern15(int row)
{   
    int count=1;
    int nst=1;
    int nsp=row/2;
    for(int r=1;r<=row;r++)
    {
        for(int csp=1;csp<=nsp;csp++)
        {
            System.out.print("\t");
        }
        //int count =r;               Yeh condition sirr ne krayi same work kregi as 
        //if(r>(row/2))
        //count=row-r+1;                            //my last line of count
       
        for(int cst=1;cst<=nst;cst++)
        {
            System.out.print(number+"\t");
            if(cst<=nst/2)
            {
                count++;
            }
            else
            {
                count--;
            }
        }
              
        if(r<=row/2)
        {   
            nst+=2;
            nsp--;
        }
        else{
            nst-=2;
            nsp++;
        }
        count=(row/2)- nsp +1;
        System.out.println();
    }
}
public static void pattern8(int row)
{
    int nsp=row-1;
      for(int r=1;r<=row;r++)
      {
          for(int csp=1;csp<=nsp;csp++)
          {
              System.out.print("\t");
          }
          System.out.println("*\t");
          nsp--;
      }

}
public static void Pattern9(int row)
{
    int nsp1=0;
    int nsp2=row/2+1;
    for(int r=1;r<=row;r++)
    {
        for(int csp=1;csp<=nsp1;csp++)
        {
            System.out.print("\t");
        }
        System.out.print("*\t");
        for(int csp=1;csp<=nsp2;csp++)
        {
            System.out.print("\t");
        }
        System.out.println("*\t");

        if(r<row/2)
        {
            nsp1++;
            nsp-=2;
        }
        else{
            nsp1--;
            nsp2+=2;
        }
    }
}

 public static void Pattern11(int row){
    int nst=1; 
    int count=1;
    for(int r=1;r<=row;r++)
    {
        for(int cst=1;cst<=nst;cst++)
        {
            System.out.print(count+"\t");
            count++;
        }
        System.out.println();
        nst++;
    }
}
    public static void pattern13(int row){
    for(int n=0;n<row;n++)
    {    
        int val=1;
        for(int r=0;r<=n;r++)
        {
            System.out.print(val+"\t");
            val=((n-r)*val)/(r+1);
        }
        System.out.println();
    }
}
public static void pattern17(int row)
    {
        int nsp=row/2,nst=1,mnst=row;
        for(int r=1;r<=row;r++)
        {   
            if(r!=row/2+1)
            {
            for(int csp=0;csp<nsp;csp++)
            {
                System.out.print("\t");
            }
            
            for(int cst=1;cst<=nst;cst++)
            {
                System.out.print("*\t");
            }
            }
            else{
                for(int cst=1;cst<=mnst;cst++)
                {
                    System.out.print("*\t");
                }
            }
            if(r<=row/2)
            {
                nst++;
            }
            else{
                nst--;
            }
            System.out.println();
        }
    }
}